sample_apps/rust/examples/utils/query_common.rs

```diff
@@ -0,0 +1,252 @@
+use aws_sdk_timestreamquery as timestream_query;
+use aws_sdk_timestreamquery::types;
+use aws_types::region::Region;
+use clap::Parser;
+use std::fs;
+use std::io::{self, Write};
+
+static DEFAULT_DATABASE_NAME: &str = "devops_multi_sample_application";
+static DEFAULT_OUTPUT_FILE: &str = "query_results.log";
+static DEFAULT_REGION: &str = "us-east-1";
+static DEFAULT_TABLE_NAME: &str = "host_metrics_sample_application";
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+pub struct Args {
+    // The Timestream for LiveAnalytics database name to use for all queries
+    #[arg(short, long, default_value = DEFAULT_DATABASE_NAME)]
+    pub database_name: String,
+
+    // The full name of the output file to write query results to.
+    // For example, query_results.log
+    #[arg(short, long, default_value = DEFAULT_OUTPUT_FILE)]
+    pub output_file: String,
+
+    // The name of the AWS region to use for queries
+    #[arg(short, long, default_value = DEFAULT_REGION)]
+    pub region: String,
+
+    // The Timestream for LiveAnalytics table name to use for all queries
+    #[arg(short, long, default_value = DEFAULT_TABLE_NAME)]
+    pub table_name: String,
+}
+
+pub async fn get_connection(
+    region: &String,
+) -> Result<timestream_query::Client, timestream_query::Error> {
+    let config = aws_config::defaults(aws_config::BehaviorVersion::latest())
+        .region(Region::new(region.to_owned()))
+        .load()
+        .await;
+    let (client, reload) = timestream_query::Client::new(&config)
+        .with_endpoint_discovery_enabled()
+        .await
+        .expect("Failure");
+    tokio::task::spawn(reload.reload_task());
+    Ok(client)
+}
+
+pub fn write(mut file: &fs::File, s: String) -> io::Result<()> {
+    let s_formatted = format!("{}\n", s);
+    file.write(s_formatted.as_bytes())?;
+    file.flush()?;
+    Ok(())
+}
+
+pub fn process_scalar_type(data: &types::Datum) -> Result<String, String> {
+    data.scalar_value
+        .clone()
+        .ok_or_else(|| "Scalar value is None".to_string())
+}
+
+pub fn process_time_series_type(
+    data: &[types::TimeSeriesDataPoint],
+    column_info: &types::ColumnInfo,
+) -> Result<String, String> {
+    let mut value = String::new();
+    for (i, datum) in data.iter().enumerate() {
+        value.push_str(&datum.time);
+        value.push(':');
+
+        let column_type = column_info.r#type();
+        let column_type_ref = column_type.as_ref().ok_or("Column type is None")?;
```

Comment:
Looks like a lot of error handling here is ignored with the `.ok_or`. Unless this expected behavior we should fail rather than set string values to errors.

Reply:
Won't this line return an `Err` if the column type is None, and the caller would have to handle the returned `Result` value?